src/public_types.ts

```diff
@@ -313,6 +313,9 @@ export interface IRepresentation {
   /** If the track is HDR, gives the HDR characteristics of the content */
   hdrInfo? : IHDRInformation;
   index : IRepresentationIndex;
+
+  /** NOTE: not part of the API. */
+  isSupported: boolean;
```

Comment:
Sadly I had to add this to properly exploit it in the debug overlay.

This has an impact on types advertised through the API, making it seems to application developers not checking the API that they can rely on this - but they cannot, it's not part of the API.

Hopefully, they shouldn't rely on weirdly named properties whose role is hard to get, without checking first what it is for first.

Reply:
Also this is not a problem in the `v4` anymore

Reply:
Yes `Representation` is an RxPlayer-only type with much more properties and `IRepresentation` is a subset of it we're forced to keep supporting in the v3 due to being part of the `getCurrentRepresentations`/`getCurrentAdaptations` etc. groups of API (removed in the v4).

I preferred rewriting clearly the subset in public_types as it generally just corresponds to what is documented in the API documentation and as such also act as a check on API breakage.

Reply:
Thanks for the reply